### PR TITLE
EDGECLOUD-3897 EDGECLOUD-3829 EDGECLOUD-3822 Client endpoints for server

### PR DESCRIPTION
**Pull request by brucearmstrong**
- Moved client directory within the Django application code, so that it can be called from an endpoint.
- New endpoints allow any deployed app inst to be used as a client for benchmarking a particular server. This lets us hit one server with multiple simultaneous low-latency clients.
-- /client/benchmark/
-- /client/download/
- Changed docker startup command to gunicorn. This allows adding or removing workers on the fly.
- utilization.py for getting real-time CPU and GPU stats.
- remote_bench.py - Simple procedural script to launch clients on multiple remote app insts at the same time.
- object_detectron2.py - Drop-in replacement for object_detector.py. Not in use, but I wanted to check it in for reference.
- consumers.py - Added FPS calculation for the websocket session.
- requirements.txt - Added some packages required by the client code.

### CODE REVIEW

File: ComputerVisionServer/moedx/client/remote_bench.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    print(response.content.decode("utf-8"))
+
+def remote_download(url):
+    # params = {'url': 'http://acrotopia.com/mobiledgex/%s' %filename}
+    params = {'url': 'http://opencv.facetraining.mobiledgex.net/videos/landscape/%s' %filename}
+    response = requests.post("%s/client/download/" %url, data=params)
+    print("Response for %s:" %url)
+    print(response.content.decode("utf-8"))
+
+urls = [
```
Comment (franklin-huang-mobiledgex) [inline ComputerVisionServer/moedx/client/remote_bench.py:32]: are you going to manually maintain this list of cv instances? Maybe we could do a GetAppInstList once in a while?

File: ComputerVisionServer/moedx/client/remote_bench.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+# filename = "objects_320x180_x2.mp4"
+filename = "objects_320x180.mp4"
+# filename = "objects.mp4"
+
+def launch_remote_benchmark(url):
+    body = '-s 80.187.140.9 -e /object/detect/ -c websocket -f %s -n PING --server-stats' %filename
+    # body = '-s cv-gpu-cluster.frankfurt-main.tdg.mobiledgex.net --tls -e /object/detect/ -c websocket -f %s -n PING --server-stats' %filename
+    # body = '-s cv-gpu-cluster.frankfurt-main.tdg.mobiledgex.net --tls -e /object/detect/ -c websocket -f %s -n PING --fullsize' %filename
+    # body = '-s cv-gpu-cluster.frankfurt-main.tdg.mobiledgex.net --tls -e /object/detect/ -c websocket -f %s -n PING' %filename
+    # body = '-s cv-gpu-cluster.frankfurt-main.tdg.mobiledgex.net --tls -e /object/detect/ -c rest -f %s -n PING --skip-frames 4' %filename
```
Comment (franklin-huang-mobiledgex) [inline ComputerVisionServer/moedx/client/remote_bench.py:20]: dead code?

File: ComputerVisionServer/moedx/client/test_dme_rest.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+# app_vers = "2020-05-19-GPU"
+# org_name = "TEF_Dev_Spain"
+
+# app_name = "mobiledgexsdkdemo"
+# carrier_name = "Telefonica"
+# app_vers = "v1"
+# org_name = "teftest"
+
+# app_name = "HA Demo"
+# org_name = "Hooli"
```
Comment (franklin-huang-mobiledgex) [inline ComputerVisionServer/moedx/client/test_dme_rest.py:43]: lol. love the Silicon valley reference

**Review by franklin-huang-mobiledgex — APPROVED** [review]
Looks good. Looks like there's a lot of commented out code though

File: ComputerVisionServer/moedx/client/remote_bench.py (hunk 2 above)
Comment (brucearmstrong) [reply ComputerVisionServer/moedx/client/remote_bench.py:20]: Keeping different versions handy to alter the tests.

**Comment by brucearmstrong** [comment]
> Looks good. Looks like there's a lot of commented out code though

Yes, I left a lot of that in the simple test scripts. Frequently used options that I want to keep handy.

File: ComputerVisionServer/moedx/client/remote_bench.py (hunk 1 above)
Comment (brucearmstrong) [reply ComputerVisionServer/moedx/client/remote_bench.py:32]: Yes, this is manual. It's set up so I can easily perform a 5-client test, a 4-client test, etc., by commenting or uncommenting code.

Passing in the number of desired clients and doing a GetAppInstList would be a nice improvement, but this benchmarking code might never get used again after I collect my final numbers.

File: ComputerVisionServer/moedx/client/test_dme_rest.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+# org_name = "Broadpeak"
+# app_vers = "02.00.rc2-2923"
+# carrier_name = "TDG"
+
+print("dme: %s, app_name: %s, org_name: %s, app_vers: %s, carrier_name: %s" %(dme, app_name, org_name, app_vers, carrier_name))
+url = "https://%s:38001/v1/registerclient" %dme
+data = {
+  "app_name": app_name,
+  "org_name": org_name,
+  # "dev_name": org_name,
```
Comment (ah1053) [inline ComputerVisionServer/moedx/client/test_dme_rest.py:62]: I don't think we need that anymore

**Review by ah1053 — APPROVED** [review]

File: ComputerVisionServer/moedx/client/remote_bench.py (hunk 1 above)
Comment (lgarner) [reply ComputerVisionServer/moedx/client/remote_bench.py:32]: Maybe say test only somewhere if leaving in?

**Review by lgarner — APPROVED** [review]
ok

**Pull request merged**